Druid-Tests/DruidTestRTLCompiler.class.st

```diff
@@ -245,15 +245,26 @@ DruidTestRTLCompiler >> gen_primitiveAdd [
 DruidTestRTLCompiler >> gen_primitiveAnd [
 	"AutoGenerated by Druid"
 
-	| jump1 jump2 currentBlock |
+	| jumpNext jumpTrue jump1 currentBlock jump2 |
 	self MoveR: ReceiverResultReg R: TempReg.
 	self CmpCq: 0 R: TempReg.
-	jump1 := self JumpLessOrEqual: 0.
+	jump1 := self JumpGreater: 0.
+	self MoveCq: 0 R: TempReg.
+	jump2 := self Jump: 0.
+	currentBlock := self Label.
+	jump1 jmpTarget: currentBlock.
 	self MoveR: ReceiverResultReg R: TempReg.
 	self CmpCq: 10 R: TempReg.
-	jump2 := self JumpGreater: 0.
+	jumpTrue := self JumpGreater: 0.
+	self MoveCq: 0 R: TempReg.
+	jumpNext := self Jump: 0.
+	jumpTrue jmpTarget: self Label.
+	self MoveCq: 1 R: TempReg.
+	jumpNext jmpTarget: self Label.
 	currentBlock := self Label.
-	jump1 jmpTarget: currentBlock.
+	jump2 jmpTarget: currentBlock.
+	self CmpCq: 1 R: TempReg.
+	jump2 := self JumpZero: 0.
 	self MoveCq: 10 R: ReceiverResultReg.
```

Comment:
Now this code is longer, maybe is related to some `DeadBranchElimination` that we lost?